src/control/build/interop.go

```diff
@@ -0,0 +1,165 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package build
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// MaxMajorDelta is the default maximum major version delta
+	// allowed between two components.
+	MaxMajorDelta = 0
+	// MaxMinorDelta is the default maximum minor version delta
+	// allowed between two components.
+	MaxMinorDelta = 2
+)
+
+// Component is a component of the system.
+type Component string
+
+// Matches returns true if the component is equal to other
+// or one of the components is a wildcard.
+func (c Component) Matches(other Component) bool {
+	return c == other || c == ComponentAny || other == ComponentAny
+}
+
+func (c Component) String() string {
+	return string(c)
+}
+
+var (
+	// ComponentAny is a wildcard component.
+	ComponentAny = Component("")
+	// ComponentServer represents the control plane server.
+	ComponentServer = Component("server")
+	// ComponentAdmin represents the Control API client.
+	ComponentAdmin = Component("admin")
+	// ComponentAgent represents the compute node agent.
+	ComponentAgent = Component("agent")
```

Comment:
Do not know if needed, but the component `"undefined"` should maybe added to have all the components from grpc_authorization.go ?

Reply:
I don't think so... In the event that an undefined component makes it through it will correctly trigger an error in `NewVersionedComponent()`.

Reply:
OK